src/assets/wise5/classroomMonitor/student-progress/student-progress.component.ts

```diff
@@ -0,0 +1,238 @@
+        const workgroupId = studentStatus.workgroupId;
+        this.updateTeam(workgroupId);
+      })
+    );
+    this.subscriptions.add(
+      this.teacherDataService.currentWorkgroupChanged$.subscribe(({ currentWorkgroup }) => {
+        this.currentWorkgroup = currentWorkgroup;
+      })
+    );
+    const context = 'ClassroomMonitor',
+      nodeId = null,
+      componentId = null,
+      componentType = null,
+      category = 'Navigation',
+      event = 'studentProgressViewDisplayed',
+      data = {};
+    this.teacherDataService.saveEvent(
+      context,
+      nodeId,
+      componentId,
+      componentType,
+      category,
+      event,
+      data
+    );
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.unsubscribe();
+  }
+
+  private initializeStudents(): void {
+    this.teams = [];
+    const workgroups = this.configService.getClassmateUserInfos();
+    for (const workgroup of workgroups) {
+      const workgroupId = workgroup.workgroupId;
+      const displayNames = this.configService.getDisplayUsernamesByWorkgroupId(workgroupId);
+      const team = {
+        periodId: workgroup.periodId,
+        periodName: workgroup.periodName,
+        workgroupId: workgroupId,
+        username: displayNames
+      };
+      this.teams.push(team);
+      this.updateTeam(workgroupId);
+    }
+    this.sortWorkgroups();
+  }
+
+  private updateTeam(workgroupId: number): void {
+    const location = this.getCurrentNodeForWorkgroupId(workgroupId);
+    const completion = this.getStudentProjectCompletion(workgroupId);
+    const score = this.getStudentTotalScore(workgroupId);
+    let maxScore = this.classroomStatusService.getMaxScoreForWorkgroupId(workgroupId);
+    maxScore = maxScore ? maxScore : 0;
+
+    for (const team of this.teams) {
+      if (team.workgroupId === workgroupId) {
+        team.location = location;
+        team.completion = completion;
+        team.score = score;
+        team.maxScore = maxScore;
+        team.scorePct = maxScore ? score / maxScore : score;
+      }
+    }
+  }
+
+  private getCurrentNodeForWorkgroupId(workgroupId: number): string {
+    return this.classroomStatusService.getCurrentNodePositionAndNodeTitleForWorkgroupId(
+      workgroupId
+    );
+  }
+
+  /**
+   * Get project completion data for the given workgroup (only include nodes with student work)
+   * @param workgroupId the workgroup id
+   * @return object with completed, total, and percent completed (integer between 0 and 100)
+   */
+  private getStudentProjectCompletion(workgroupId: number): any {
+    return this.classroomStatusService.getStudentProjectCompletion(workgroupId, true);
+  }
+
+  private getStudentTotalScore(workgroupId: number): number {
+    return this.teacherDataService.getTotalScoreByWorkgroupId(workgroupId);
+  }
+
+  private sortWorkgroups(): void {
+    this.sortedTeams = [];
+    for (const team of this.teams) {
+      this.sortedTeams.push(team);
+    }
```

Comment:
Is this the same (but with fewer lines)?
```suggestion
    this.sortedTeams = [...this.teams];
```